apps/roles.go

```diff
@@ -0,0 +1,124 @@
+package apps
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/evergreen-ci/gimlet"
+)
+
+// Role is the data structure used to read and manipulate user roles in these routes
+type Role struct {
+	ID          *string           `json:"id"`
+	Name        *string           `json:"name"`
+	ScopeType   *string           `json:"scope_type"`
+	Scope       *string           `json:"scope"`
+	Permissions map[string]string `json:"permissions"`
```

Comment:
what is this mapping?

Reply:
It's a list of things that this role has access to for this resource. I'm expecting the keys in the map to be defined by the consuming app and not an end user